tests/gtfs/report/test_report_utils.py

```diff
@@ -0,0 +1,105 @@
+"""Test scripts for the GTFS report utility functions."""
+
+import os
+import pathlib
+import re
+
+import pytest
+from pyprojroot import here
+
+from transport_performance.gtfs.report.report_utils import (
+    TemplateHTML,
+    set_up_report_dir,
+)
+
+
+@pytest.fixture(scope="function")
+def template_fixture():
+    """Fixture for test funcs expecting a valid feed object."""
+    template = TemplateHTML(
+        path=here("tests/data/gtfs/report/html_template.html")
+    )
+    return template
+
+
+class TestTemplateHTML(object):
+    """Tests related to the TemplateHTML class."""
+
+    def test_init(self, template_fixture):
+        """Test initialising the TemplateHTML class."""
+        expected_template = """<!DOCTYPE html>
+<html lang="en">
+
+<body>
+    <div>[test_placeholder] Tester [test_placeholder]</div>
+</body>
+"""
+        assert (
+            expected_template == template_fixture.get_template()
+        ), "Test template not as expected"
+
+    def test_insert_defence(self, template_fixture):
+        """Test defences for .insert()."""
+        with pytest.raises(
+            ValueError,
+            match=(
+                "You have selected not to replace multiple"
+                "placeholders of the same value, however"
+                "placeholders occur more than once. \n"
+                "If you would like to allow this, set the"
+                "replace_multiple param to True"
+            ),
+        ):
+            template_fixture.insert("test_placeholder", "test_value")
+
+    def test_insert_on_pass(self, template_fixture):
+        """Test functionality for .insert() when defences are passed."""
+        expected_template = """<!DOCTYPE html>
+<html lang="en">
+
+<body>
+    <div>test_value Tester test_value</div>
+</body>
+"""
+        template_fixture.insert(
+            placeholder="test_placeholder",
+            value="test_value",
+            replace_multiple=True,
+        )
+        assert (
+            expected_template == template_fixture.get_template()
+        ), "Test placeholder replacement not acting as expected"
+
+
+class TestSetUpReportDir(object):
```

Comment:
Tests in this class interact with test/data/gtfs/report . Best practice to avoid writing out lint on running test suite. Instead, use `pytest.tmpdir` fixture to create the temp dir structure you want to work with. Once the test run completes, any tmps will be cleaned up. An example implementation is in tests`/gtfs/test_gtfs_utils.py test_bbox_filter_gtfs_writes_as_expected`